Dockerfile.fs

```diff
@@ -0,0 +1,18 @@
+FROM golang:1.15-buster AS build
+
+RUN apt-get update && apt-get install -y --no-install-recommends bash curl git
+RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+
+WORKDIR /go/src/github.com/analogj/lodestone-publisher/
+
+ADD Gopkg.toml ./
+ADD Gopkg.lock ./
+ADD cmd ./cmd
+ADD pkg ./pkg
+
+RUN dep ensure
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o lodestone-fs-publisher ./cmd/fs-publisher/fs-publisher.go
+
+FROM scratch
+COPY --from=build /go/src/github.com/analogj/lodestone-publisher/lodestone-fs-publisher /lodestone-fs-publisher
+CMD ["/lodestone-fs-publisher"]
```

Comment:
Quick question here:

The current storage container that links to the prebuilt-binary uses the a command `/lodestone-fs-publisher start ...` with some args (see https://github.com/LodestoneHQ/lodestone-storage/blob/fa67ffb379c418590b47b747a3a22b310411e9f2/rootfs/etc/services.d/fs-publisher/run#L7)

When I tried earlier this week to do the same thing you did here, the filesystem publisher would basically run its default help function and output a bunch of stuff to the console, then exit. I don't see any changes to the actual Go cmd in this PR that would change that behavior. Do we need to change the `CMD` line to execute a shell script to start the FS publisher the same way it's started in the existing `lodestone-storage` container?

Reply:
I don't think that is a good idea. Those parameters are configuration. Exactly what you should be passing from the outside (docker-compose in the main repo).

It is more customary to use environment variables over CLI args for native docker applications though. I expect we can change it in the future. But if we decide to do it that way, we should not do it by a wrapper script or default command in docker. We should instead make the application itself to read the environment variables directly. It is fairly easy given the choice of libraries used in the project. They already support it, it is just matter of enabling that properly and maybe restructuring the variables a bit. It would be nice to unify them across all the apps, so that you can reuse them and not have to worry that one application wants something in form of URL, the other wants it as separate hostname and port, etc.

Regarding those wrapper scripts, they are almost always a bad idea. They usually limit the flexibility in some way - ie. what if you want to add one more option? What if you want to override existing argument which was magically predefined for you? What if you want to base your image on it and add more stuff to it for your integration purposes? In all those cases you usually end up overriding the command and trying to mimic the previous default behaviour in some way. It is much better to have sane and transparent configuration interface directly on the app.

Reply:
Gotcha...fair enough. But my question still stands though. Does the fs-publisher remain persistent and watch the folders using `CMD ["/lodestone-fs-publisher"]` or is it going to just print its "usage" info to the log and exit since the command didn't include the `start` argument like the current way its invoked does.

Reply:
Yes, it will output the usage info and that is a correct behaviour. It has no way of doing his thing without a configuration to help it connect with the rest. That is where this comes in: https://github.com/LodestoneHQ/lodestone/pull/127/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R86

Reply:
Ah, thank you! I had totally glossed over that part of the compose file. Apologies for the run-around.